sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileTable.scala

```diff
@@ -22,23 +22,27 @@ import org.apache.hadoop.fs.FileStatus
 
 import org.apache.spark.sql.{AnalysisException, SparkSession}
 import org.apache.spark.sql.execution.datasources._
-import org.apache.spark.sql.sources.v2.{DataSourceOptions, SupportsBatchRead, SupportsBatchWrite, Table}
+import org.apache.spark.sql.sources.v2.{SupportsBatchRead, SupportsBatchWrite, Table}
 import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
 
 abstract class FileTable(
     sparkSession: SparkSession,
-    options: DataSourceOptions,
+    options: CaseInsensitiveStringMap,
+    paths: Seq[String],
```

Comment:
Hi, @cloud-fan .
Should we change `FileTable` signature to accept `paths` additionally for merging `DataSourceOptions` and `CaseInsensitiveStringMap`?

Reply:
it's not a big deal. I did this because we need paths in the `OrcDataSourceV2` as well, so we can calculate the paths only once in the `OrcDataSourceV2`.

Reply:
Got it. Thanks!